src/main/java/edu/kis/powp/jobs2d/drivers/MacroDriver.java

```diff
@@ -0,0 +1,39 @@
+package edu.kis.powp.jobs2d.drivers;
+
+import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.command.CompoundCommand;
+import edu.kis.powp.jobs2d.command.DriverCommand;
+import edu.kis.powp.jobs2d.command.OperateToCommand;
+import edu.kis.powp.jobs2d.command.SetPositionCommand;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MacroDriver implements Job2dDriver {
+    private List<DriverCommand> macro = new ArrayList<>();
+
+    @Override
+    public void operateTo(int x, int y) {
+        macro.add(new OperateToCommand(x, y));
+    }
+
+    @Override
+    public void setPosition(int x, int y) {
+        macro.add((new SetPositionCommand(x, y)));
+    }
+
+    public CompoundCommand getMacro(){
+        return new CompoundCommand(macro, "Macro");
+    }
```

Comment:
I have commented this previously, but the comment seemed to vanish (maybe it wasn't saved). This is not a deep copy. Check and apply one of the features introduced by another group:
- A visitor for a deep copy of a complex command
- A deep copy of a complex command